### DISCUSSION TITLE
[SOLVED] strongswan 5.9.9(charon-svc) + Windows faild to run

### DISCUSSION THREAD

**Discussion by psalty** [strongswan/strongswan#1535] [Q&A]
Hi. I tried to build strongswan 5.9.9 and run on Windows.

However, when I tried to run charon-src.exe I faced following errors.

```
00[DMN] Starting IKE service charon-svc (strongSwan 5.9.9, Windows Client 6.2.9200 (SP 0.0)
00[LIB] feature CUSTOM:libcharon in critical plugin 'charon-svc' has unmet dependency: NONCE_GEN
00[LIB] feature CUSTOM:libcharon-receiver in critical plugin 'charon-svc' has unmet dependency: HASHER:HASH_SHA1
00[LIB] feature CUSTOM:libcharon-sa-managers in critical plugin 'charon-svc' has unmet dependency: HASHER:HASH_SHA1
00[LIB] failed to load 3 critical plugin features
```

my strongswan.conf is

charon-svc {
	load_modular = yes
  	start-scripts {
    		swanctl-creds = swanctl --load-creds --noprompt
    		swanctl-conns = swanctl --load-conns
  	}
	plugins {
		include strongswan.d/charon/*.conf
	}
}

include strongswan.d/*.conf

My openssl build was 1.1.1n, and my configure options for strongswan is  follows

```
            --disable-defaults --enable-monolithic --enable-static \
            --enable-svc --enable-ikev2 --enable-ikev1 \
            --enable-nonce --enable-pem --enable-pkcs1 \
            --enable-openssl \
            --enable-socket-win \
            --enable-kernel-wfp --enable-kernel-iph \
            --enable-swanctl --with-swanctldir=swanctl \
            --enable-x509 \
            --enable-pubkey \
            --enable-pkcs12 \
            --exec-prefix=/opt/tmp \
            --prefix=/opt/tmp \
            --sbindir=/opt/tmp/bin \
            --sysconfdir=/opt/tmp/etc \
            --libdir=/opt/tmp/lib \
            --with-strongswan-conf=strongswan.conf --host=x86_64-w64-mingw32
```
Is there any step that I missed? I have never seen this issue on other platforms.

Thanks.

**Comment by psalty** [comment]
I think I found the answer.
it turns out the plugins directories should be included in strongswan.conf
like ./plugins/*.conf

**Reply by tobiasbrunner** [reply]
That's what `include strongswan.d/charon/*.conf` does if those directories and files exist (`strongswan.d` should be in the same directory as strongswan.conf). You can also disable `load_modular` to just use the compiled-in plugin list (see [here](https://docs.strongswan.org/docs/5.9/plugins/pluginLoad.html) for details).

**Comment by cchangwen** [comment]
`net stop IKEEXT` (IKE and AuthIP IPsec Keying Modules), and run as administrator.
